lib/hl7/datetime_parser.rb

```diff
@@ -0,0 +1,62 @@
+require "hl7/datetime_components"
+require "active_support/core_ext/module/delegation"
+
+module HL7
+  class DatetimeParser
+    # The HL7 datetime format is:
+    #    YYYY[MM[DD[HH[MM[SS[.S[S[S[S]]]]]]]]][+/-ZZZZ]
+    RX = /\A(\d{4})(?:(\d{2})(?:(\d{2})(?:(\d{2})(?:(\d{2})(?:(\d{2})(\.\d{1,4})?)?)?)?)?)?([-+]\d{4})?\Z/.freeze
+
+    def initialize(text)
+      @text = text
+    end
+
+    def components
+      @components ||= (
+        y, mo, d, h, mi, s, f, off = match
+
+        DatetimeComponents.new(
+          year: y.to_i,
+          month: (mo.nil? ? 1 : mo.to_i),
+          day: (d.nil? ? 1 : d.to_i),
+          hour: h.to_i,
```

Comment:
Why the nil checks on `mo` and `d` but nothing else?

Reply:
I'm relying on the fact that `nil.to_i => 0`, which is the default value for those fields (as opposed to month and day, which are 1-indexed). But that's probably too subtle, and I should be more explicit about the defaults to make the code easier to read.

Reply:
> But that's probably too subtle, and I should be more explicit about the defaults to make the code easier to read.

I agree